services/QuillLMS/spec/models/user_spec.rb

```diff
@@ -2007,12 +2007,21 @@
       teacher.save
     end
 
-    it 'should create new TeacherInfo record' do
-      teacher = build(:teacher)
+    context 'should create new TeacherInfo record' do
+
+      it 'should create a record' do
+        teacher = build(:teacher)
+
+        expect do
+          teacher.save
+        end.to change(TeacherInfo, :count).by(1)
+      end
+
+      it 'should save the role_selected_at_signup attribute to TeacherInfo record' do
+        teacher = create(:teacher, role_selected_at_signup: 'admin')
+        expect(teacher.teacher_info.role_selected_at_signup).to eq(teacher.role_selected_at_signup)
+      end
```

Comment:
I think there's some (very vague) risk of this not behaving as expected.  If the `create` call was somehow bugged and set `role_selected_at_signup` on the model to `nil`, this could pass even if it's not doing what we want.  I think I'd set the and check the constant directly:
```ruby
let(:role_selected_at_signup) { 'admin' }
let(:teacher) { create(:teacher, role_selected_at_signup:)

it { expect(teacher.teacher_info.role_selected_at_signup).to eq(role_selected_at_signup) }
```